openstreetmap-carto.lua

```diff
@@ -515,99 +750,134 @@ function split_tags(tags, tag_map)
     return cols
 end
 
-
--- TODO: Make add_* take object, not object.tags
-function add_point(tags)
+-- Functions to add point, line, polygon and route objects to database
+function add_point(object)
+    local tags = object.tags
     local cols = split_tags(tags, columns_map.point)
     cols['layer'] = layer(tags['layer'])
     tables.point:add_row(cols)
 end
 
-function add_line(tags)
+function add_line(object)
+    local tags = object.tags
     local cols = split_tags(tags, columns_map.line)
     cols['layer'] = layer(tags['layer'])
-    cols['z_order'] = z_order(tags)
-    cols.way = { create = 'line', split_at = 100000 }
+    if add_z_order_col then
+        cols['z_order'] = z_order(tags)
+    end
+    cols.way = { create = 'line', split_at = max_length }
     tables.line:add_row(cols)
 end
 
-function add_roads(tags)
+function add_roads(object)
+    local tags = object.tags
     local cols = split_tags(tags, columns_map.roads)
     cols['layer'] = layer(tags['layer'])
-    cols['z_order'] = z_order(tags)
-    cols.way = { create = 'line', split_at = 100000 }
+    if add_z_order_col then
+        cols['z_order'] = z_order(tags)
+    end
+    cols.way = { create = 'line', split_at = max_length }
     tables.roads:add_row(cols)
 end
 
-function add_polygon(tags)
+function add_polygon(object)
+    local tags = object.tags
     local cols = split_tags(tags, columns_map.polygon)
     cols['layer'] = layer(tags['layer'])
-    cols['z_order'] = z_order(tags)
-    cols.way = { create = 'area', multi = true }
+    if add_z_order_col then
+        cols['z_order'] = z_order(tags)
+    end
+    cols.way = { create = 'area'}
+    if not multi_geometry then
+        cols.way.split_at = 'multi'
+    end    
     tables.polygon:add_row(cols)
 end
 
 function add_route(object)
+    local tags = object.tags
     for i, member in ipairs(object.members) do
         if member.type == 'w' then
-            local cols = split_tags(object.tags, columns_map.roads)
+            local cols = split_tags(tags, columns_map.line)
             cols.member_id = member.ref
             cols.member_position = i
+            cols.route = tags['route']            
+            cols.ref = tags['ref']
+            cols.network = tags['network']
```

Comment:
I've fixed this upstream, but this fix is wrong. split_tags needs to be called on the columns of the route_member table, and tests needed adding to catch this case.

Reply:
Ok, thanks, yes, this was one of the harder parts, as although I get the gist of what you're trying to do here, I wasn't entirely sure about the technical implementation, so it's no real surprise if I got it wrong.

Reply:
> **split_tags** needs to be called on the columns of the route_member table

@pnorman , I've read through the [osm2pgsql manual](https://osm2pgsql.org/doc/manual.html), and especially the flex sections, but I don't see a **split_tags** helper function mentioned. Is this an undocumented new feature?

I am not sure what to do with this at this point in time, do you have a clear suggestion for how to fix this?

Reply:
I have realized, that my last question was superfluous, it is a custom function in the style itself that I overlooked.